tests/snappi_tests/multidut/bgp/files/bgp_outbound_helper.py

```diff
@@ -0,0 +1,1157 @@
+            logger.info('Skipping Router id for {}, Since bgp is not configured'.
+                        format(topology.DeviceGroup.find()[0].Name))
+            continue
+    logger.info('\n')
+    logger.info('Testing with Route Range: {}'.format(route_range))
+    logger.info('\n')
+    for i in range(0, iteration):
+        logger.info(
+            '|--------------------------- Iteration : {} -----------------------|'.format(i+1))
+        logger.info("Starting all protocols ...")
+        ps = api.protocol_state()
+        ps.state = ps.START
+        api.set_protocol_state(ps)
+        wait(TIMEOUT, "For Protocols To start")
+        logger.info('Verifying protocol sessions state')
+        protocolsSummary = StatViewAssistant(ixnetwork, 'Protocols Summary')
+        protocolsSummary.CheckCondition('Sessions Down', StatViewAssistant.EQUAL, 0)
+        logger.info('Starting Traffic')
+        ts = api.transmit_state()
+        ts.state = ts.START
+        api.set_transmit_state(ts)
+        wait(TIMEOUT, "For Traffic To start")
+
+        flow_stats = get_flow_stats(api)
+        port_stats = get_port_stats(api)
+
+        logger.info('\n')
+        logger.info('Rx Snappi Port Name : Rx Frame Rate')
+        for port_stat in port_stats:
+            if int(port_stat.name.split('_')[-1]) >= t2_port_index_start-1:
+                logger.info('{} : {}'.format(port_stat.name, port_stat.frames_rx_rate))
+                pytest_assert(port_stat.frames_rx_rate > 0, '{} is not receiving any packet'.format(port_stat.name))
+        logger.info('\n')
+        for i in range(0, len(traffic_type)):
+            logger.info('{} Loss %: {}'.format(flow_stats[i].name, int(flow_stats[i].loss)))
+            pytest_assert(int(flow_stats[i].loss) == 0, f'Loss Observed in {flow_stats[i].name} before link Flap')
+
+        sum_t2_rx_frame_rate = 0
+        for port_stat in port_stats:
+            if int(port_stat.name.split('_')[-1]) >= t2_port_index_start:
+                sum_t2_rx_frame_rate = sum_t2_rx_frame_rate + int(port_stat.frames_rx_rate)
+        # Flap the required test port
+        if duthosts[0].hostname == flap_details['device_name']:
+            logger.info(' Shutting down {} port of {} dut !!'.
+                        format(flap_details['port_name'], flap_details['device_name']))
+            duthosts[0].command('sudo config interface shutdown {} \n'.
+                                format(flap_details['port_name']))
+        elif 'Ixia' == flap_details['device_name']:
+            ixn_port = ixnetwork.Vport.find(Name=flap_details['port_name'])[0]
+            ixn_port.LinkUpDn("down")
+            logger.info('Shutting down snappi port : {}'.format(flap_details['port_name']))
+        wait(TIMEOUT, "For link to shutdown")
+        flow_stats = get_flow_stats(api)
+        delta_frames = 0
+        for i in range(0, len(traffic_type)):
+            delta_frames = delta_frames + flow_stats[i].frames_tx - flow_stats[i].frames_rx
```

Comment:
Can we check the frames delta, when tx=rx?

Also lets check the convergence time when the links are UP.

Reply:
implemented